### ISSUE TITLE
Please explicitly specify the coordinate system.

### ISSUE THREAD

**Issue by pmcesky** [napari/napari#1904]
## 📚 Documentation

I find it is really frustrating working with coordinates in Napari. I don't see any explicit information on the coordinate system and coordinate order that napari used. Is it always in TZYX, or ZYX, or YX format across different layers? Could this info be added to the documentation and tutorial? Every time I write coordinates (like creating points, specifying Scale), I need to confirm I write them in the correct way, as my data are usually in another format. Please clarify this. Thanks!

**Comment by sofroniewn** [comment]
HI @pmcesky sorry you find working with coordinates so frustrating. By default coordinates are rendered in reserve order, so a convention like `TZYX` for all layers. We'll make that more explicit in the documentation.

Are things working for you now, or are you still running into problems? I'd be happy to try and help if the later.

**Comment by Czaki** [comment]
I this moment there is no glue between TZYX mnemonics and coordinate system. Napari does not standardize the order of the axis. And different file formats have different orders (even inside tiff there could different orders of axis storage). 

No @sofroniewn. There is no place to provide data axis order. Also, default, napari readers, as I know, does not standardized axis order. 

The most common is TZYX, but three is no guarantee. And there is a big problem with C Axes. Because the file could have TZYXC (ome-tiff), TZCYX (ImageJ tiff). Other orders are also available. 

This is a much bigger problem and partially connected with units problem #1701.

**Comment by pmcesky** [comment]
Hi @sofroniewn
Thanks very much for this clarification!
Now I'm much more confident to use this great app.

**Comment by sofroniewn** [comment]
Don't hesitate to reach out again if you have questions, or spot places where our documentation or tutorials can be improved. We're still a work in progress, but glad you're trying napari out!!

**Comment by pmcesky** [comment]
> Don't hesitate to reach out again if you have questions, or spot places where our documentation or tutorials can be improved. We're still a work in progress, but glad you're trying napari out!!

Yes! I recognize napari as my No.1 image viewer currently. Though sometimes things may be confusing, I'd like to see it goes forward and improves. I'm glad if I could help some.

**Comment by d-v-b** [comment]
The xarray solution to this is to use named axes everywhere. It adds overhead when specifying a transform (you have to provide a dict instead of and array, e.g. `translation = {'x' : 0, 'y': 0, 'z': 0}` instead of `translation = [0,0,0]`) but it completely eliminates the ambiguity described in this issue.

**Comment by jni** [comment]
> Also, default, napari readers, as I know, does not standardized axis order.
>
> The most common is TZYX, but three is no guarantee. And there is a big problem with C Axes. Because the file could have TZYXC (ome-tiff), TZCYX (ImageJ tiff). Other orders are also available.

This is correct: we don't have any axis order. We display arrays in the order they are given, with the last few dimensions being displayed.

At the risk of being unpopular :joy:, I would go further, though: I want *by design* to never have standardised axis ordering. If you want to use data in TZCYX, napari will do it, if you want to do XYZ, napari will also do it. We will always want to work with bare NumPy arrays, which precludes named axes everywhere — even if we give defaults of {0, 1, 2, ...}, this will break NumPy broadcasting if applied uniformly. (NumPy broadcasting rules right-align array dimensions.) @sofroniewn ran into this and at one point gave default axis names of {..., -3, -2, -1}. *That* works with NumPy broadcasting but is entirely unintuitive, although that might be the best solution — happy to revisit.

We *could* definitely build something that would transpose datasets by default *if* they are already annotated with axis names *and* the viewer itself is also annotated. I think that would satisfy most use cases while keeping the viewer general enough for bare NumPy arrays. How does that sound @d-v-b @pmcesky @Czaki?

**Comment by sofroniewn** [comment]
> This is correct: we don't have any axis order. We display arrays in the order they are given, with the last few dimensions being displayed.

Yes, well put, but see below

> No @sofroniewn. There is no place to provide data axis order. Also, default, napari readers, as I know, does not standardized axis order.

You can use the `viewer.dims.order` to determine which axes are displayed, but that isn't settable via reader plugins yet as it still needs to be worked on see #1077.

**Comment by Czaki** [comment]
> You can use the `viewer.dims.order` to determine which axes are displayed, but that isn't settable via reader plugins yet as it still needs to be worked on see #1077.

But it is still connected with the internal axis order. Not with mnemonic names. If you have two tiff files. One with order TZCYX and the second one with TZYXC. Then the current loading method will create confusion. Setting `viewer.dims.order` does not will solve this.

Also, tifffile package by default squash empty axes. So if you have two files one with data on TYX axes and the second with data on ZYX axes then they will be represented identically. Even when loaded simultaneously.  And if napari takes care of axis order from the file metadata it should cause showing two sliders (one for time and one for z stack), not one as it is currently. 

@jni But I agree with @jni that there should be no strong glue to axes mnemonics. Nice will be such support from default readers. Maybe possibility to set names for axes in `viewer.dims` and pass it to layers which should allow getting common experience.

**Comment by d-v-b** [comment]
>At the risk of being unpopular 😂, I would go further, though: I want by design to never have standardised axis ordering. If you want to use data in TZCYX

I agree, and I would be even more unpopular and say that 90% of the time "C" should not be an axis, but rather the number of elements returned when sampling from a point in space :)

>We could definitely build something that would transpose datasets by default if they are already annotated with axis names and the viewer itself is also annotated. 

I like this idea!

**Comment by melissawm** [comment]
In the docs wg meeting we talked about adding a docs page on "spatial coordinates and dimensions in napari", not to be prescriptive but to explain what we have as built-in settings and how people can explore the gui to change them.

**Comment by jni** [comment]
My current issue is #4633 and the fact that we actually want to swap the default handedness of the space. There is of course value in documenting that it's currently left-handed, but I think I would prefer spending effort on flipping the leading axis so that we are right-handed by default, *then* documenting things.

**Comment by DragaDoncila** [comment]
@jni totally see what you're saying, but will note that #4633 has been open already for more than two years, and this request is almost four years old. 

Even if we do end up changing things to address #4633, lots of users might be on older versions and have questions about the way things used to work. I don't think we should hold off on documenting how these things currently work just because we'll **eventually** change it. I'd much rather go to an existing document and edit/add to it, than write up a document from scratch.

It'd be different if there was a PR waiting to be merged. I think this information is too critical to not have documented somewhere...

**Comment by melissawm** [comment]
Is this connected to https://github.com/napari/docs/issues/529?

**Comment by jni** [comment]
It can be, but not necessarily, since napari/docs#529 can be done exclusively in 2D space (and would probably benefit from that, to keep complexity down). Having said that, even in that case, we probably do want to specify that the origin is at the top left and the axes point down, and then right, in that order. And that pixel centers are at integer coordinates. (see e.g. the second screenshot in #1660.)
